### PR TITLE
Get started basics

### PR DESCRIPTION
**Pull request by dvsrepo**

### CODE REVIEW

**Comment by dvsrepo** [comment]
Needs a lot of proof-reading but I would prefer to merge it and we will revise it in the coming days, share it with our collaborators, etc.

**Comment by dvsrepo** [comment]
But if you have some time to do a quick proof reading, we can get some changes in

File: src/biome/text/configuration.py [hunk 1]
```diff
@@ -58,10 +58,10 @@ def from_params(
         cls: Type["FeaturesConfiguration"], params: Params, **extras
     ) -> "FeaturesConfiguration":
 
-        word = params.pop("word", None)
+        word = params.pop("word", params.pop("words", None))  # TODO: remove backward
```
Comment (frascuchon) [inline src/biome/text/configuration.py:61]: I think this is a bad merge/rebase commit

File: src/biome/text/configuration.py [hunk 2]
```diff
@@ -58,10 +58,10 @@ def from_params(
...
         word = WordFeatures(**word.as_dict(quiet=True)) if word else None
 
-        char = params.pop("char", None)
+        char = params.pop("char", params.pop("chars", None))  # TODO: remove backward
```
Comment (frascuchon) [inline src/biome/text/configuration.py:64]: same as above

File: src/biome/text/configuration.py [hunk 3]
```diff
@@ -189,21 +189,15 @@ class PipelineConfiguration(FromParams):
     features: `FeaturesConfiguration`
         The input `features` to be used by the model pipeline. We define this using a `FeaturesConfiguration` object.
     head: `TaskHeadConfiguration`
-        The `head` for the task, e.g., a LanguageModelling task, using a `TaskHeadConfiguration` object.
+        The `head` for the task, e.g., a LanguageModelling task, using a `TaskHeadSpec` object.
```
Comment (frascuchon) [inline src/biome/text/configuration.py:192]: same comment as above

File: src/biome/text/configuration.py [hunk 4]
```diff
@@ -189,21 +189,15 @@ class PipelineConfiguration(FromParams):
...
     tokenizer: `TokenizerConfiguration`, optional
         The `tokenizer` defined with a `TokenizerConfiguration` object.
-    encoder: `Encoder`
-        The core text seq2seq `encoder` of our model using a `Seq2SeqEncoderConfiguration`
+    encoder: `Seq2SeqEncoderSpec`
```
Comment (frascuchon) [inline src/biome/text/configuration.py:195]: Same comment as above

File: src/biome/text/configuration.py [hunk 5]
```diff
@@ -189,21 +189,15 @@ class PipelineConfiguration(FromParams):
...
+        The core text seq2seq `encoder` of our model using a `Seq2SeqEncoderSpec`
```
Comment (frascuchon) [inline src/biome/text/configuration.py:196]: Same comment as above

File: src/biome/text/configuration.py [hunk 6]
```diff
@@ -352,31 +346,31 @@ def __init__(
...
-    Use this to configure a Vocabulary using specific arguments from `allennlp.data.Vocabulary``
+    Use this to configure a Vocabulary using specific arguments from allennlp.data.Vocabulary.
 
     See [AllenNLP Vocabulary docs](https://docs.allennlp.org/master/api/data/vocabulary/#vocabulary])
 
     Parameters
     ----------
     sources: `List[DataSource]`
         Datasource to be used for data creation
-    min_count: `Dict[str, int]`, optional (default=None)
```
Comment (frascuchon) [inline src/biome/text/configuration.py:363]: @dvsrepo   Do you remove the optional defaults for any particular reason?

File: src/biome/text/configuration.py [hunk 7]
```diff
@@ -352,31 +346,31 @@ def __init__(
...
+    min_count: `Dict[str, int]`
         Minimum number of appearances of a token to be included in the vocabulary.
         The key in the dictionary refers to the namespace of the input feature.
-    max_vocab_size:  `Union[int, Dict[str, int]]`, optional (default=`None`)
```
Comment (frascuchon) [inline src/biome/text/configuration.py:366]: Same comment as above

File: src/biome/text/configuration.py [hunk 8]
```diff
@@ -352,31 +346,31 @@ def __init__(
...
+    max_vocab_size:  `Union[int, Dict[str, int]]`
         Maximum number of tokens in the vocabulary
-    pretrained_files: `Optional[Dict[str, str]]`, optional
```
Comment (frascuchon) [inline src/biome/text/configuration.py:368]: Same comment as above

File: src/biome/text/configuration.py [hunk 9]
```diff
@@ -352,31 +346,31 @@ def __init__(
...
+    pretrained_files: `Optional[Dict[str, str]]`
         If provided, this map specifies the path to optional pretrained embedding files for each
         namespace. This can be used to either restrict the vocabulary to only words which appear
         in this file, or to ensure that any words in this file are included in the vocabulary
         regardless of their count, depending on the value of `only_include_pretrained_words`.
         Words which appear in the pretrained embedding file but not in the data are NOT included
         in the Vocabulary.
-    only_include_pretrained_words: `bool`, optional (default=False)
```
Comment (frascuchon) [inline src/biome/text/configuration.py:375]: Same comment as above

File: src/biome/text/configuration.py [hunk 10]
```diff
@@ -352,31 +346,31 @@ def __init__(
...
-        Only include tokens present in pretrained_files
-    tokens_to_add: `Dict[str, int]`, optional
```
Comment (frascuchon) [inline src/biome/text/configuration.py:377]: Same comment as above

File: src/biome/text/configuration.py [hunk 11]
```diff
@@ -352,31 +346,31 @@ def __init__(
...
+    only_include_pretrained_words: `bool`
+        Only include tokens from pretrained_files
+    tokens_to_add: `Dict[str, int]`
         A list of tokens to add to the vocabulary, even if they are not present in the ``sources``
-    min_pretrained_embeddings: ``Dict[str, int]``, optional
```
Comment (frascuchon) [inline src/biome/text/configuration.py:379]: Same comment as above

**Comment by dvsrepo** [comment]
@frascuchon  closing this, ignore it and see https://github.com/recognai/biome-text/pull/320

**Pull request closed without merging**